### PR TITLE
fix: replace imgur to s3 for various files without conflict

### PR DESCRIPTION
**Pull request by ahmaxed**
(cherry picked from commit 4af8ae9cc4301abb299a0bc91e959f64c4be62d7)

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Reference #36007

### CODE REVIEW

**Review by raisedadead — APPROVED** [review]
Hey @freeCodeCamp/moderators ! 

Can we delegate the QA for this PR to anyone who is interested? I am approving on behalf of dev-team to enable the merge.

Thanks for your help.

**Comment by RandellDawson** [comment]
I will take this one.

**Review by RandellDawson — APPROVED** [review]
@raisedadead  I have validated the url conversions are correct on this PR.  Do we want at least two mods to approve before merging?

**Comment by tchaffee** [comment]
I would suggest instead of hard-coding `https://cdn-media-1.freecodecamp.org` that it come from the `.env` file or something similar. Then it will never need to be changed in code again. It also makes it possible for the code to be used by different orgs, and to make it easier for the site to be used offline, which @QuincyLarson mentioned as a goal.

**Comment by RandellDawson** [comment]
@raisedadead I was planning on reviewing the other 5 PRs which replace imgur images, but I will now wait until you respond to @tchaffee's question above.  Just let me know what you want to do.   If you do go the IMAGE_BASE_URL route, you will have to rewrite any`testString` regex to use a RegExp constructor to recreate all of the current regex expressions.  So you go from a `testString` which currently looks like this:
```yml
    testString: 'assert(code.match(/background:\s*?url\(\s*("|"|)https:\/\/cdn-media-1\.freecodecamp\.org\/imgr\/MJAkxbh\.png\1\s*\)/gi));`
```
to something that looks like this:
```yml
    testString: 'const escapedUrl = IMAGE_BASE_URL.replace(/([/.])/g, '\\$1'); const regex = new RegExp("background:\\s\*\?url\\(\\s\*\("|"|\)" + escapedUrl + "imgr\/MJAkxbh\\.png\\1\\s\*\\)", "gi"); assert(code.match(regex));'
```
If there are many testString using these urls, then I would recommend using a package called [escape-string-regexp](https://www.npmjs.com/package/escape-string-regexp) and then you use the library to help avoid some of these complex regex's which would be required to craft.

Either way, this makes things a bit more complicated for these challenge files unless there is an easier approach I am missing.

**Comment by ojeytonwilliams** [comment]
Regarding the guide markdown, I think we'd need to create a remark plugin to insert `IMAGE_BASE_URL`.  I would be happy to try and write one, if that's the plan, but it may take a little while.

**Comment by raisedadead** [comment]
I agree with Todd here, although these PRs should go in first and the `.env` updates can come in a later PR to keep the contribution going. So, @RandellDawson  please go ahead and QA the PRs.

We should also put up a separate issue that can be deferred until later about the `.env` setup. This is because, it would also cover a lot of the other hard coded URLs and values.

**Comment by RandellDawson** [comment]
@raisedadead I have approved this PR already.  Do you want at least two approvals before merge?  If so, I will notify the other active mods.

File: docs/style-guide-for-guide-articles.md
Comment (thecodingaviator) [inline docs/style-guide-for-guide-articles.md]: @raisedadead We'll need to see what to do with this. We obviously cannot change it since users cannot upload to fCC's CDN afaik

Comment (RandellDawson) [reply docs/style-guide-for-guide-articles.md]: Good catch!

File: curriculum/challenges/english/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.english.md
Comment (thecodingaviator) [inline curriculum/challenges/english/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.english.md]: Do we need to change this, I believe we can just enter any URL there, the current one doesnot even make sense as a BG

File: docs/style-guide-for-guide-articles.md
Comment (raisedadead) [reply docs/style-guide-for-guide-articles.md]: I believe the changes are accidentally made with search and replace @ahmadabdolsaheb?

Can you double check all these again.

File: curriculum/challenges/english/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.english.md
Comment (ojeytonwilliams) [reply curriculum/challenges/english/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.english.md]: I think it's fine as is.  The URL does have to be an image for the camper to see a change.  Also, this image does create the subtle change that the instructions talk about.

**Review by ojeytonwilliams — COMMENTED** [review]
Aside from the style guide issue @thecodingaviator spotted, everything LGTM.

File: docs/style-guide-for-guide-articles.md
Comment (ahmadabdolsaheb) [reply docs/style-guide-for-guide-articles.md]: @thecodingaviator Nice catch.
@raisedadead yes some of them were find and replace. I will look into them.

Comment (ahmadabdolsaheb) [reply docs/style-guide-for-guide-articles.md]: Probably we need to change these instructions in the future to prevent campers from uploading to imgur all together.

File: curriculum/challenges/english/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.english.md
Comment (ahmadabdolsaheb) [reply curriculum/challenges/english/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.english.md]: @thecodingaviator, thanks for the catch. If it is alright with you, I would rather differ content modifications specially in the curriculum to another PR since it would require a further discussion.

**Comment by ahmadabdolsaheb** [comment]
I double checked this. it should be good to go.

File: docs/style-guide-for-guide-articles.md
Comment (thecodingaviator) [reply docs/style-guide-for-guide-articles.md]: @ahmadabdolsaheb We'll need to discuss that. Please open an issue

File: curriculum/challenges/arabic/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.arabic.md
Comment (thecodingaviator) [inline curriculum/challenges/arabic/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.arabic.md]:
```suggestion
<section id="instructions"> باستخدام رابط من <code>https://i.imgur.com/MJAkxbh.png</code> ، تعيين <code>background</code> من صفحة كاملة مع <code>body</code> محدد. </section>
```

File: curriculum/challenges/arabic/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.arabic.md
Comment (thecodingaviator) [inline curriculum/challenges/arabic/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.arabic.md]:
```suggestion
    testString: 'assert(code.match(/background:\s*?url\(\s*("|"|)https:\/\/i\.imgur\.com\/MJAkxbh\.png\1\s*\)/gi));'
```

File: curriculum/challenges/english/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.english.md
Comment (thecodingaviator) [inline curriculum/challenges/english/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.english.md]:
```suggestion
Using the url of <code>https://i.imgur.com/MJAkxbh.png</code>, set the <code>background</code> of the whole page with the <code>body</code> selector.
```

File: curriculum/challenges/english/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.english.md
Comment (thecodingaviator) [inline curriculum/challenges/english/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.english.md]:
```suggestion
    testString: assert(code.match(/background:\s*?url\(\s*("|'|)https:\/\/i\.imgur\.com\/MJAkxbh\.png\1\s*\)/gi));
```

File: curriculum/challenges/portuguese/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.portuguese.md
Comment (thecodingaviator) [inline curriculum/challenges/portuguese/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.portuguese.md]:
```suggestion
<section id="instructions"> Usando o url de <code>https://i.imgur.com/MJAkxbh.png</code> , defina o <code>background</code> de toda a página com o seletor de <code>body</code> . </section>
```

File: curriculum/challenges/portuguese/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.portuguese.md
Comment (thecodingaviator) [inline curriculum/challenges/portuguese/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.portuguese.md]:
```suggestion
    testString: 'assert(code.match(/background:\s*?url\(\s*("|"|)https:\/\/i\.imgur\.com\/MJAkxbh\.png\1\s*\)/gi));'
```

File: curriculum/challenges/spanish/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.spanish.md
Comment (thecodingaviator) [inline curriculum/challenges/spanish/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.spanish.md]:
```suggestion
<section id="instructions"> Usando la url de <code>https://i.imgur.com/MJAkxbh.png</code> , establezca el <code>background</code> de toda la página con el selector de <code>body</code> . </section>
```

File: curriculum/challenges/spanish/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.spanish.md
Comment (thecodingaviator) [inline curriculum/challenges/spanish/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.spanish.md]:
```suggestion
    testString: 'assert(code.match(/background:\s*?url\(\s*("|"|)https:\/\/i\.imgur\.com\/MJAkxbh\.png\1\s*\)/gi));
```

**Review by thecodingaviator — CHANGES_REQUESTED** [review]
I believe we'll consequently need to update these in this PR

File: curriculum/challenges/english/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.english.md
Comment (thecodingaviator) [reply curriculum/challenges/english/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.english.md]: Since we're doing it to the Egnlish file, we'll need to do it to the other languages too. If you want to do it in this PR, fine enough, I'm totally fine if you want another file

Comment (ahmadabdolsaheb) [reply curriculum/challenges/english/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.english.md]: Thank you @thecodingaviator, I took note of suggestions, and I will address them in a separate PR.

Comment (ahmadabdolsaheb) [reply curriculum/challenges/english/01-responsive-web-design/applied-visual-design/create-texture-by-adding-a-subtle-pattern-as-a-background-image.english.md]: If everything else other than changing the background look acceptable, please feel free to merge this PR.

**Review by thecodingaviator — APPROVED** [review]
Other than the curriculum files, I guess everything looks good to me, so I'm going to merge this! @ahmadabdolsaheb Please go through with the PR we talked about

**Comment by thecodingaviator** [comment]
Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 📝

**Pull request merged**
